### PR TITLE
Improve resolution of docker sha256 to docker image name in some cases

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

We've seen issues in some cases, where we don't tag properly when image/inspect does not have sufficient information to retrieve a valid image name.

Solution is to add a fallback to .Config.Image in case first image/inspect did not work.
Note that it may still trigger some tagging changes (especially for `docker_image` tag if image/inspect answers differently over time).

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/util/containers/image.go [hunk 1]
```diff
@@ -10,6 +10,13 @@ import (
 	"strings"
 )
 
+var (
+	// ErrEmptyImage is returned when argument is empty
```
Comment (xornivore) [inline pkg/util/containers/image.go]: Nit - image name is empty?
```suggestion
	// ErrEmptyImage is returned when image name argument is empty
```

File: pkg/util/containers/image.go [hunk 2]
```diff
@@ -10,6 +10,13 @@ import (
...
+	ErrEmptyImage = errors.New("empty image name")
+	// ErrImageIsSha256 is returned when argument is a not an image name but a sha256
```
Comment (xornivore) [inline pkg/util/containers/image.go]:
```suggestion
	// ErrImageIsSha256 is returned when image name argument is a sha256
```

**Review by xornivore — DISMISSED** [review]
👍

File: pkg/tagger/collectors/docker_extract_test.go [hunk 3]
```diff
@@ -348,6 +348,26 @@ func TestDockerExtractImage(t *testing.T) {
...
 			},
 		},
+		{
+			testName: "Some Nomad Setup",
+			co: types.ContainerJSON{
+				ContainerJSONBase: &types.ContainerJSONBase{
+					Image: "sha256:380b233f1574da39494e2b36e65f262214fe158af5ae7a94d026b7a4e46fa358",
+				},
+				Config: &container.Config{
+					Image: "quay.io/hashicorp/cloud-consul-ama:3451-be4c56f",
```
Comment (martinb3) [inline pkg/tagger/collectors/docker_extract_test.go]: Hi there! Is it possible to use more anonymized repos/orgs here and below? Thanks!

Comment (vboulineau) [reply pkg/tagger/collectors/docker_extract_test.go]: Of course, done!

**Pull request merged**